lib/index.coffee

```diff
@@ -0,0 +1,134 @@
+_ = require "underscore"
+_.mixin require "underscore.deep"
+async = require "async"
+genericPool = require "generic-pool"
+debug = require("debug")("thrift-pool")
+
+# create_cb creates and initializes a connection
+#  @param thrift, used to create connection
+#  @param pool_options, host and port used to create connection
+#  @param thrift_options, passed to thrift connection,
+#
+create_cb = (thrift, pool_options, thrift_options, cb) ->
+  cb = _.once cb
+  connection = thrift.createConnection pool_options.host, pool_options.port, thrift_options
+  connection.__ended = false
+  connection.on "connect", ->
+    debug "in connect callback"
+    connection.connection.setKeepAlive(true)
+    cb null, connection
+  connection.on "error", (err) ->
+    debug "in error callback"
+    connection.__ended = true
+    cb err
+  # "close" should only be called after a "connect" event
+  connection.on "close", ->
+    debug "in close callback"
+    connection.__ended = true
+  # timeout listener only applies if timeout is passed into thrift_options
+  # if "timeout" emits, it sets the connection as ended
+  if thrift_options.timeout?
+    debug "adding timeout listener"
+    connection.on "timeout", ->
+      debug "in timeout callback"
+      connection.__ended = true
```

Comment:
`cb err` here? I'm actually not sure, though part of me thinks it couldn't hurt.